File: crates/ferric-parse/tests/fixtures/sample.rs
//! Sample Rust file for ferric-parse integration tests.

pub struct Calculator {
    value: f64,
}

pub enum Operation {
    Add,
    Multiply,
}

pub fn standalone_fn(x: f64) -> f64 {
    x * 2.0
}

impl Calculator {
    pub fn new(value: f64) -> Self {
        Self { value }
    }

    pub fn add(&self, other: f64) -> f64 {
        self.value + other
    }
}

File: crates/ferric-parse/tests/fixtures/sample.py
"""Sample Python file for ferric-parse integration tests."""


class Calculator:
    """A simple calculator."""

    def add(self, a: int, b: int) -> int:
        """Add two numbers."""
        return a + b

    def multiply(self, a: int, b: int) -> int:
        return a * b


def standalone_function(x: float) -> float:
    """A standalone function."""
    return x * 2.0
